test/integration/dependency.bats

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function teardown() {
+        swarm_manage_cleanup
+        stop_docker
+}
+
+@test "shared volumes dependency" {
+        start_docker 2
+        swarm_manage
+
+        # Running the second container with shared volumes.
+        run docker_swarm run --name b1 -e constraint:node==node-0 -d busybox:latest sleep 500
+        [ "$status" -eq 0 ]
+        run docker_swarm run --name b2 --volumes-from=/b1 -d busybox:latest sh
+        [ "$status" -eq 0 ]
+
+        # check if containers share volume.
+        run docker_swarm inspect -f "{{ .HostConfig.VolumesFrom }}" b2
+	[ "$status" -eq 0 ]
+        [[ "${output}" == *"[/b1]"* ]]
+
+	# check if container "b2" is started on the same node 
```

Comment:
I know this is a bit redundant since we are forcing `node==node-0` for b1, but could you actually check if `b1` is running on `node-0` as well? It will also make the test easier to understand.